enhancements/ibm-public-cloud/ibm-public-cloud-support.md

```diff
@@ -0,0 +1,243 @@
+- machine-config-operator
+- cloud-credential-operator
+- cluster-authentication-operator
+
+Components/Changes needed for this to work:
+
+#### OpenShift ROKS Toolkit CLI
+
+CLI tool that takes a release image and a configuration file as input and produces a set
+of manifests that provision a control plane that runs on an existing Kubernetes cluster.
+PKI certificates are not generated by the tool. Manifests of control plane components 
+expect certificates to have been generated before provisioning and to exist as secrets
+and configmaps on the control plane namespace following a predefined contract.
+
+This tool is invoked by IBM Public Cloud provisioning code to when instantiating a new cluster.
+
+#### Control Plane Operator
+An operator that:
+- Projects global configuration into the target cluster. Watches
+  and overwrites any changes to global configuration initiated inside the user
+  cluster.
+- Ensures that the Kubernetes Controller Manager CA bundle includes self-signed CAs
+  generated by the router and the service CA operator.
+
+#### Cluster Version Operator Changes
+In phase I the cluster version operator excludes operator manifests that
+have an annotation in the form:
+
+`exclude.release.openshift.io/[identifier]=true`
+
+whenever a global variable in the form
+
+`EXCLUDE_MANIFESTS=[identifier]`
+
+is present. For IBM public cloud, the value of \[identifier\] is 
+`internal-openshift-hosted`
+
+#### Excluded Manifests
+Manifests for the components that should be skipped by the CVO whenever
+the `EXCLUDE_MANIFESTS=internal-openshift-hosted` environment variable is 
+present, should include the annotation to exclude them. In most cases, only the
+minimum set of manifests to allow skipping the component should be annotated.
+However, in the case of the Machine API and Machine Configuration operators, 
+the CRDs that represent machines, machinesets and autoscalers should also be
+skipped. Monitoring alerts for components that do not get installed in the user
+cluster should also be skipped where possible.
+
+#### Console Changes
+The console should not report the control plane as being down if no metrics
+datapoints exist for control plane components in this configuration.
+
+### GA
+
+For GA, there is more native support for IBM Public Cloud as a supported
+cloud provider in the product. Operators for control plane
+components (based on the control plane self-hosted operators) are deployed on 
+the management cluster.
+
+Components and changes required for this phase:
+
+#### Control Plane Operators
+Break up the single Beta control plane operator into different operators that
+vendor code from the self-hosted control plane operators in the product. The vendored
+code should include config observers that assemble a new configuration for their
+respective control plane components. This will ensure that drift in future versions
+is kept under control and that a single code base is used to manage control plane
+configuration.
```

Comment:
where is this beta control plane operator?

Reply:
we're working on it, will be added to the current hypershift-toolkit repo. For the second phase we will create separate repos for each of the control plane controllers.